### PR TITLE
New resource: azurerm_private_dns_zone

### PR DESCRIPTION
**Pull request by timja**
Fixes https://github.com/terraform-providers/terraform-provider-azurerm/issues/3714

### CODE REVIEW

File: azurerm/resource_arm_private_dns_zone_test.go [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+			},
+			{
+				Config:      testAccAzureRMPrivateDnsZone_requiresImport(ri, location),
+				ExpectError: testRequiresImportError("azurerm_private_dns_zone"),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMPrivateDnsZone_withTags(t *testing.T) {
```
Comment (timja) [inline azurerm/resource_arm_private_dns_zone_test.go]: This test is currently failing for me locally.

The tags get correctly applied but don't seem to be in the state, not sure what is going on here.

Any help appreciated

Comment (timja) [reply azurerm/resource_arm_private_dns_zone_test.go]: It's happening because the response from the API only has the old values, its returning stale data.

Comment (timja) [reply azurerm/resource_arm_private_dns_zone_test.go]: I've added a sleep, unless anyone has any better ideas...

**Comment by bentterp** [comment]
This is good! 
I just can't figure out how to use it for VNET auto-registrations. Do we have another resource type coming for that, or am I just not getting it? Tried creating a link in the portal and then generating a plan, no changes required.

**Comment by timja** [comment]
It needs a new resource type, no issue created for it yet AFAIK.
I just created the first resource required for private dns.

**Comment by bentterp** [comment]
#3758 :-D

I'm quite good at creating issues, no good at writing the code for the PR's....

**Pull request merged**